journal/202306.md

```diff
@@ -0,0 +1,861 @@
+- [Updated](https://github.com/decred/dcrd/commit/6c323c18b0eb7e4035c8ceb4ac3a36230c8c9c42) the `getwork` command and `notifywork` websocket notification. These are used by the mining code to fetch new data for hashing.
+- Added the [ASERT difficulty calculation](https://github.com/decred/dcrd/commit/d9b1921177d4626a22c380853d6c7030a2d5c854) and algorithm for miners to [generate blocks](https://github.com/decred/dcrd/commit/2db55eb6e7d16aca5676cd7d473555d23b30e23f) with the correct difficulty.
+- Implemented the DCP-11 [consensus vote](https://github.com/decred/dcrd/commit/85e3701a67fa5a9a7eb418ce71fa991ede447dbb). It will show up as `blake3pow` in voting software.
+
+In addition to the BLAKE3 PoW hash change which will make current Decred ASIC miners useless, the new ASERT algorithm will adjust the difficulty with each new block instead of every 144 blocks (the current adjustment frequency). For reference, Bitcoin's mining difficulty is updated every 2,016 blocks, or roughly every two weeks. ASERT stands for Absolutely Scheduled Exponentially weighted Rising Targets.
+
+The ASERT difficulty algorithm is also exponential, and can handle surging activity much better than the existing algorithm. This will greatly disincentivize "hit and run" miners who are only looking for a quick reward at a low difficulty. ASERT also better handles [large drops of hashrate](https://github.com/decred/dcrd/pull/3115#discussion_r1205741807) which cause slow blocks and degraded user experience. The network is stronger when miners are consistently creating new blocks and maintain the target block time of 5 minutes. See also the relevant [discussion on Matrix](https://matrix.to/#/!uCnoeYfbhUnvwWihzS:decred.org/$tbF9LyO1EzQ8m3JDHfX9EYnQz1zr8W0Ra73Yjtprj1M). For those who love fast math and bit tricks, make sure to read the [ASERT code here](https://github.com/decred/dcrd/blob/d9b1921177d4626a22c380853d6c7030a2d5c854/blockchain/standalone/pow.go#L222).
+
+{ _link to DCP-11 if available_ }
+
+{ _do we have the stats how much faster ASERT will adjust to a sharp spike/sharp drop of hashpower, compared to EMA?_ }
+
+{ _what will be the difficulty value immediately after the reset? to make it tangible, what computing power will require to mine, say how many average modern CPUs or GPUs?_ }
```

Comment:
It's in the [chain params](https://github.com/decred/dcrd/blob/a13ca3ace61d28678e1889b9a6303dd88c3dae98/chaincfg/mainnetparams.go#L115). 

Concretely, that is `0x1b00a5a6` which equates to an initial target hash rate of about 1.45 TH/s.  The number of GPUs majorly depends on which GPUs and the speeds that can be obtained by the implementation used to power them.

It doesn't really matter much though because it will very quickly adjust due to the new algorithm and the fact it adjusts every block.

Reply:
Thanks for sharing this hashrate number. Looks low enough to continue the chain with even very low initial hashrate. I [added](https://github.com/xaur/decred-news/commit/b147200935c98ab6443ec68e43bef8f687e4854e#diff-1d487d8831a0f2f7b2e5f0ea1543054a4e3b287dd31e36d115af3525a053b485R109) both the number and a mention that it is not very important due to how fast ASERT is. But it's still interesting to mention IMO.